src/client/app/components/admin/users/CreateUserModalComponent.tsx

```diff
@@ -0,0 +1,201 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react';
+import { useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Alert, Button, Col, Container, FormFeedback, FormGroup, Input, Label, Modal,
+	ModalBody, ModalFooter, ModalHeader, Row } from 'reactstrap';
+import { userApi } from '../../../redux/api/userApi';
+import { User, UserRole } from '../../../types/items';
+import { showErrorNotification, showSuccessNotification } from '../../../utils/notifications';
+import translate from '../../../utils/translate';
+
+/**
+ * Defines the create user modal form
+ * @returns CreateUserModal component
+ */
+export default function CreateUserModal() {
+
+	// user default values
+	const userDefaults = {
+		email: '',
+		password: '',
+		confirmPassword: '',
+		role: '',
+		note: '',
+		passwordMatch: true
```

Comment:
Yes, I have decided to remove it.

Reply:
I am unsure this should go away.  I feel like it has to be checked if either password and confirmPassword changes to ensure they are matching. Otherwise if you only checked after confirmPassword, the user could go back to password and change it and there would be no check. useEffect seems the simplest method to handle this. Let me know if I am overthinking or missing something.

Reply:
Given the added useEffect and additional checks I think it is okay to leave as is.